nav2_costmap_2d/include/nav2_costmap_2d/layered_costmap.hpp

```diff
@@ -197,7 +213,7 @@ class LayeredCostmap
   bool isOutofBounds(double robot_x, double robot_y);
 
 private:
-  Costmap2D costmap_;
+  Costmap2D plugins_costmap_, costmap_;
```

Comment:
This is not a good name - we should figure out another name for these costmaps to separate them. Maybe `master_costmap_` for the main one? I'm not sure I have a better name suggestion for the plugins costmap, though it could potentially use one to be more clear what it is on a casual read-through

Reply:
Not finally sure about `master_costmap_`, as both costmaps by their meaning are master costmap. One of them is the basis/basement/bottom layer for the second final costmap. So, how about to rename `plugins_costmap_` -> to `basic_costmap_`, `preliminary_costmap_` or `bottom_costmap_`?

Reply:
`combined_costmap_` for the combined filter/plugins and `plugin_costmap_` for just the plugins?

Add a comment in the header file about why there are two and what each are for. That should help stem some of the confusion later.

Reply:
In this case, I'd still consider to remain `bottom/preliminary/basic_costmap_` and `costmap_` (or `combined_costmap_`) variables. Having `plugin_costmap_` and `combined_costmap_` indeed can cause some confusion: if there is no filters enabled we will use `combined_costmap_` directly only for the plugins, but if one plugin was enabled, we will write all plugins' work into `plugin_costmap_`. In other words, for the plugins `plugin_costmap_` won't always be used.
So, now I see most reasonable to have to variables:  `bottom/preliminary/basic_costmap_` for bottom layer and `costmap_` or  `combined_costmap_` for final result. What do you think about it?

Reply:
How about `primary_costmap_` for the plugins and  `combined_costmap_` for combined / final?

Reply:
Agree with this naming. Changes were applied in https://github.com/ros-planning/navigation2/pull/2308/commits/9c1c7d319f8940e1953990e7606cbf66463651d6.